### ISSUE TITLE
Ability to derive osmids and tag information from extra_info (extended storages)

### ISSUE THREAD

**Issue by TimMcCauley** [GIScience/openrouteservice#371]

**Comment by rabidllama** [comment]
what do you mean by "tag information"? Storing the osm id can be done easily as there is already the osmid extended storage that just needs a bit of modification to optimise storage size, and then that id could be used by a client to get tag data from osm itself. The issue there however would be if the data is changed on osm after the graph has been built, there will be a difference between osm data and that used in the graph

**Comment by TimMcCauley** [comment]
The idea was that the user would maybe like to see the osm_id(s) of a tollway and what kind of tollway it is.

**Comment by rabidllama** [comment]
well the id thing should be relatively straightforward. providing the type of tollway may take a bit longer though as currently it just returns whether it is a tollway or not, and so the return vlaues would need changing to represent the different types. It seems that the info is stored in the extended storage though

**Comment by nilsnolde** [comment]
Do we really need that extra info of individual attributes? I feel like OSM ID is sufficient to derive all relevant attributes from that segment.

**Comment by aoles** [comment]
> It seems that the info is stored in the extended storage though

That's correct, the different types of tollways are stored according to [`TollwayType`](https://github.com/GIScience/openrouteservice/blob/development/openrouteservice/src/main/java/heigit/ors/routing/graphhopper/extensions/TollwayType.java#L20-L38). These values are then matched against the routing request such that only tollway information relevant to a given profile is returned. So in a sense it is not really necessary to return additional tollway type details. @TimMcCauley could you maybe clarify what use case you had in mind?

**Comment by TimMcCauley** [comment]
@aoles even if they are matched towards the request, a user may want to see what types of tollways they are. 

Use case: As a logistics planner using my own maps I want to visualise the toll road according to its toll way type, e.g. https://wiki.openstreetmap.org/wiki/Talk:Key:toll / https://wiki.openstreetmap.org/wiki/DE:Key:toll

**Comment by aoles** [comment]
Thanks @TimMcCauley! We could then return the [`TollwayType`](https://github.com/GIScience/openrouteservice/blob/development/openrouteservice/src/main/java/heigit/ors/routing/graphhopper/extensions/TollwayType.java#L20-L38) id instead of a binary tollway flag as it is now. This should be relatively easy to implement both at the backend side, as well as in the app. Feel free to open a dedicated issue if you think this idea makes sense.

**Comment by TimMcCauley** [comment]
Yup, sounds like a reasonable approach - low priority though.

**Comment by MichaelsJP** [comment]
There is always the possibility to get the OSM ID in the response. 
By now, we have enough tools in the ecosystem to derive any kind of metadata from other sources by OSM ID and a simple script.
No need to replicate things like [ohsome-api](https://docs.ohsome.org/ohsome-api/v1/) or [overpass-turbo]( https://overpass-turbo.eu/)
